controllers/update_test.go

```diff
@@ -1123,40 +1157,53 @@ func mockSignature(time time.Time) *git2go.Signature {
 	}
 }
 
-func clone(repoURL, remoteName, branchName string) (*git2go.Repository, error) {
+func clone(repoURL, branchName string) (*git2go.Repository, error) {
 	dir, err := os.MkdirTemp("", "iac-clone-*")
 	if err != nil {
 		return nil, err
 	}
+	transportOptsURL := "http://" + randStringRunes(5)
+	managed.AddTransportOptions(transportOptsURL, managed.TransportOptions{
+		TargetURL: repoURL,
+	})
+	defer managed.RemoveTransportOptions(transportOptsURL)
+
 	opts := &git2go.CloneOptions{
 		Bare:           false,
 		CheckoutBranch: branchName,
 		CheckoutOptions: git2go.CheckoutOptions{
 			Strategy: git2go.CheckoutForce,
 		},
-		FetchOptions: git2go.FetchOptions{
-			RemoteCallbacks: git2go.RemoteCallbacks{
-				CertificateCheckCallback: func(cert *git2go.Certificate, valid bool, hostname string) error {
-					return nil
-				},
-			},
-		},
 	}
+	repo, err := git2go.Clone(transportOptsURL, dir, opts)
 
-	return git2go.Clone(repoURL, dir, opts)
+	// set the origin remote url to the actual repo url, since
+	// the origin remote will have transportOptsURl as the it's url
+	// because that's the url used to clone the repo.
+	err = repo.Remotes.SetUrl("origin", repoURL)
+	if err != nil {
+		return nil, err
+	}
+	return repo, nil
 }
 
 func waitForNewHead(g *WithT, repo *git2go.Repository, branch, preChangeHash string) {
 	var commitToResetTo *git2go.Commit
 
+	cleanup, err := configureManagedTransportOptions(repo, "")
+	if err != nil {
+		panic(err)
+	}
+	defer cleanup()
+
+	origin, err := repo.Remotes.Lookup("origin")
+	if err != nil {
+		panic("origin not set")
```

Comment:
Don't think we need to introduce any panic. Just `g.Expect()` assertions should be enough to stop further execution.